src/sql/platform/telemetry/telemetry.contribution.ts

```diff
@@ -13,42 +13,152 @@ import { IStorageService, StorageScope } from 'vs/platform/storage/common/storag
+			let lastMonthDate = new Date(monthlyLastUseDate);
+			this.sendUsageEvent(monthlyUseCount, lastMonthDate);
+
+			const wasActiveLastMonth: boolean = storageService.getBoolean('telemetry.wasActiveLastMonth', StorageScope.GLOBAL, false);
+
+			if (firstTimeUser) {
+				// new user
+				this.sendGrowthTypeEvent(UserGrowthType.NewUser, lastMonthDate);
+			}
+
+			// continuing or returning user
+			this.sendGrowthTypeEvent(wasActiveLastMonth ? UserGrowthType.ContinuingUser : UserGrowthType.ReturningUser, lastMonthDate);
+
+			// set wasActiveUserLastMonth
+			storageService.store('telemetry.wasActiveLastMonth', true, StorageScope.GLOBAL);
+
+			// reset the monthly count for the new month
+			storageService.store('telemetry.monthlyUseCount', 1, StorageScope.GLOBAL);
+			storageService.store('telemetry.monthlyLastUseDate', todayString, StorageScope.GLOBAL);
+		} else {
+			// if it's the same month, increment the monthly use count
+			storageService.store('telemetry.monthlyUseCount', monthlyUseCount+1, StorageScope.GLOBAL);
+		}
+	}
+
+	private didDayChange(lastUseDateNumber: number): boolean {
+		let nowDateNumber: number = Date.parse(new Date().toUTCString());
+		if (this.diffInDays(nowDateNumber, lastUseDateNumber) >= 1) {
+			return true;
+		} else {
+			let nowDate = new Date(nowDateNumber);
+			let lastUseDate = new Date(lastUseDateNumber);
+			return nowDate.getUTCDay() !== lastUseDate.getUTCDay();
+		}
+	}
+
+	private didWeekChange(lastUseDateNumber: number): boolean {
+		let nowDateNumber: number = Date.parse(new Date().toUTCString());
+		if (this.diffInDays(nowDateNumber, lastUseDateNumber) >= 7) {
+			return true;
+		} else {
+			let nowDate = new Date(nowDateNumber);
+			let lastUseDate = new Date(lastUseDateNumber);
+			return nowDate.getUTCDay() < lastUseDate.getUTCDay();
 		}
+	}
 
+	private didMonthChange(lastUseDateNumber: number): boolean {
+		let nowDateNumber: number = Date.parse(new Date().toUTCString());
+		if (this.diffInDays(nowDateNumber, lastUseDateNumber) >= 30) {
+			return true;
+		} else {
+			let nowDate = new Date(nowDateNumber);
+			let lastUseDate = new Date(lastUseDateNumber);
+			return nowDate.getUTCMonth() !== lastUseDate.getUTCMonth();
+		}
 	}
 
 	private diffInDays(nowDate: number, lastUseDate: number): number {
-		return (nowDate - lastUseDate) / (24 * 3600 * 1000);
+		return (nowDate - lastUseDate) / (3600 * 1000 * 24);
+	}
+
+	// Usage Metrics
+	private sendUsageEvent(monthlyUseCount: number, lastMonthDate: Date): void {
+		let userUsageType: UserUsageType;
+		if (monthlyUseCount === 1) {
+			userUsageType = UserUsageType.TireKicker;
+		} else if (monthlyUseCount >= 2 && monthlyUseCount <= 11) {
+			userUsageType = UserUsageType.Occasional;
+		} else if (monthlyUseCount >= 12 && monthlyUseCount <= 20) {
+			userUsageType = UserUsageType.Engaged;
+		} else if (monthlyUseCount > 20) {
+			userUsageType = UserUsageType.Dedicated;
+		}
+		this.telemetryService.publicLog('telemetry.userUsage',
```

Comment:
To stop the build failures from the strict null checks you should only send the event if userUsageType is not null (and then default it to null)